Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -8,10 +8,19 @@
 
 import SourceKittenFramework
 
+public enum RuleKind: String {
+    case lint
+    case idiomatic
+    case style
+    case metrics
+    case performance
+}
+
 public protocol Rule {
     init() // Rules need to be able to be initialized with default values
     init(configuration: Any) throws
     static var description: RuleDescription { get }
+    static var kind: RuleKind { get }
```

Comment:
Rules with configurations that make the rule be a different category depending on what config is given.